c7n/resources/asg.py

```diff
@@ -492,6 +492,49 @@ def process(self, asgs, event=None):
         return super(VpcIdFilter, self).process(asgs)
 
 
+@filters.register('failing')
+class FailingFilter(Filter):
+    schema = type_schema(
+        'failing', count={'type': 'number'})
+
+    def asg_failed_launch(self, activities={}, token=None):
+        limit_date = datetime.now(tz=tzutc()) - timedelta(days=1)
+        client = local_session(
+            self.manager.session_factory).client('autoscaling')
+        if token:
+            results = client.describe_scaling_activities(NextToken=token)
+        else:
+            results = client.describe_scaling_activities()
+        if 'NextToken' in results:
+            token = results['NextToken']
+
+        for result in results['Activities']:
+            status_code = result['StatusCode']
+            event_time = result['StartTime']
+            if status_code != 'Failed' or event_time.date() < limit_date.date():
```

Comment:
from what i can tell, the activity results are sorted by event time (most recent event first).